libr/include/r_util/r_uleb128.h

```diff
@@ -5,9 +5,9 @@
 // compression used to store an arbitrarily large integer in a small number of
 // bytes. LEB128 is used in the DWARF debug file format.
 
-R_API const ut8 *r_uleb128(const ut8 *data, int datalen, ut64 *v);
+R_API const ut8 *r_uleb128(const ut8 *data, int datalen, ut64 *v, const char **error);
```

Comment:
Is it really necessary to return a full heap-allocated string? What about:
```suggestion
typedef enum {
   R_ULEB_UNDEFINED_SHIFT_ERR,
   R_ULEB_MALFORMED_ERR,
} RUlebError;
R_API const ut8 *r_uleb128(const ut8 *data, int datalen, ut64 *v, RUlebError *error);
```

and let the caller just choose what to print exactly? You have to think these functions *may* be called by other callers as well and it's not said all of them actually want those strings you have provided.

Reply:
> Is it really necessary to return a full heap-allocated string?

Most definitely not.

> and let the caller just choose what to print exactly? You have to think these functions _may_ be called by other callers as well and it's not said all of them actually want those strings you have provided.

The idea was that if someone passes something other than NULL in `error` he knows what he's doing and what should he do if `error` is not NULL when the utility function returns.

I like your idea and if you think that it will be better in any way than the one currently suggested, I will go for it.

Reply:
@thestr4ng3r @trufae WDYT?

Reply:
Yes, i think it's fine this way. At least as an intermediate step for another refactoring, because as @phakeobj said, there are many other calls that may not be checking for errors and this should be done in another PR.